staker/block_validator.go

```diff
@@ -52,25 +50,55 @@ var (
 	validatorMsgCountLastValidationSentGauge = metrics.NewRegisteredGauge("arb/validator/msg_count_last_validation_sent", nil)
 )
 
+type LegacyValidatorInstance interface {
+	CheckLegacyValid() error
+	LegacyLastValidatedInfo() (*legacyLastBlockValidatedDbInfo, error)
+	SetLegacyValidInfo(info *legacyLastBlockValidatedDbInfo)
+}
+
+// ValidatorInstance is an interface used by BlockValidator to carry out block and MEL-state validations. BlockValidatorInstance implements
+// this interface to carry out block validations and MELValidatorInstance (TODO) implements this to carry out MEL-state validation
+type ValidatorInstance interface {
+	// Methods corresponding to ValidatorInstance's overall context
+	IsValidatedGlobalStateNew(gs validator.GoGlobalState) bool
+	LastValidatedGlobalState() validator.GoGlobalState
+	SetLastValidatedGlobalState(gs validator.GoGlobalState)
+	LastValidatedCount() (count uint64, isCaughtup bool, err error)                     // returns last validated count and a boolean indicating if the chain has caught up to that count
+	LatestProcessedMessageCount() (uint64, error)                                       // latest processed but not validated message count
```

Comment:
How about revisiting those func comments in this interface?
Most of them are redundant, or almost redundant, and could be removed if the func and return names are a little bit more extensive.